api/src/test/java/com/epam/pipeline/controller/user/UserControllerTest.java

```diff
@@ -0,0 +1,504 @@
+
+    @Autowired
+    private UserApiService mockUserApiService;
+
+    @Autowired
+    private AuthManager mockAuthManager;
+
+    @Test
+    @WithMockUser
+    public void shouldGetSettings() {
+        doReturn(token).when(mockUserApiService).issueToken(TEST_STRING, ID);
+
+        final MvcResult mvcResult = performRequest(get(USER_TOKEN_URL)
+                .params(multiValueMapOf(EXPIRATION, ID,
+                                        NAME, TEST_STRING)));
+
+        verify(mockUserApiService).issueToken(TEST_STRING, ID);
+        assertResponse(mvcResult, token, SecurityCreatorUtils.JWT_RAW_TOKEN_TYPE);
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldGetSettingsForCurrentUser() {
+        doReturn(token).when(mockAuthManager).issueTokenForCurrentUser(ID);
+
+        final MvcResult mvcResult = performRequest(get(USER_TOKEN_URL)
+                .params(multiValueMapOf(EXPIRATION, ID,
+                                        null, null)));
+
+        verify(mockAuthManager).issueTokenForCurrentUser(ID);
+        assertResponse(mvcResult, token, SecurityCreatorUtils.JWT_RAW_TOKEN_TYPE);
+    }
+
+    @Test
+    public void shouldFailGetSettings() {
+        performUnauthorizedRequest(get(USER_TOKEN_URL));
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldGetCurrentUser() {
+        doReturn(pipelineUser).when(mockUserApiService).getCurrentUser();
+
+        final MvcResult mvcResult = performRequest(get(WHOAMI_URL));
+
+        verify(mockUserApiService).getCurrentUser();
+        assertResponse(mvcResult, pipelineUser, UserCreatorUtils.PIPELINE_USER_INSTANCE_TYPE);
+    }
+
+    @Test
+    public void shouldFailGetCurrentUser() {
+        performUnauthorizedRequest(get(WHOAMI_URL));
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldRedirectWithCookieRouteType() throws Exception {
+        doReturn(token).when(mockAuthManager).issueTokenForCurrentUser(null);
+
+        final MvcResult mvcResult = performRequest(get(ROUTE_URL)
+                .params(multiValueMapOf(URL, TEST_STRING,
+                                        TYPE, RouteType.COOKIE)),
+                                        TEXT_HTML_UTF8_CONTENT_TYPE);
+
+        verify(mockAuthManager).issueTokenForCurrentUser(null);
+        assertThat(mvcResult.getResponse().getContentAsString()).contains(redirectCookie);
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldRedirectWithFormRouteType() throws Exception {
+        doReturn(token).when(mockAuthManager).issueTokenForCurrentUser(null);
+
+        final MvcResult mvcResult = performRequest(get(ROUTE_URL)
+                .params(multiValueMapOf(URL, TEST_STRING,
+                                        TYPE, RouteType.FORM)),
+                                        TEXT_HTML_UTF8_CONTENT_TYPE);
+
+        verify(mockAuthManager).issueTokenForCurrentUser(null);
+
+        assertThat(mvcResult.getResponse().getContentAsString()).contains(redirectForm);
+    }
+
```

Comment:
The controller method `/route` is one of a very few methods accessible by so-called anonymous users. Currently it is the only one if I'm correct. Could you please add the test that checks that a user with a single role `ROLE_ANONYMOUS_USER` without even `ROLE_USER` can access the method.